includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -707,14 +707,15 @@ public function get_charge_object( $charge_id = '', $params = [] ) {
 	 * @since 7.0.2
 	 * @param object $intent Stripe API Payment Intent object response.
 	 *
-	 * @return object
+	 * @return string|object
 	 */
 	public function get_latest_charge_from_intent( $intent ) {
 		if ( ! empty( $intent->charges->data ) ) {
 			return end( $intent->charges->data );
-		} else {
+		} elseif ( ! empty( $intent->latest_charge ) ) {
```

Comment:
This just fixes a warning shown at the top of the page for pre-orders, the `latest_charge` property does not exist in this case (per my tests). Also, [can be nullable per Stripe's documentation](https://docs.stripe.com/api/payment_intents).